src/node_modules/logger.js

```diff
@@ -308,94 +351,81 @@ export default class Logger extends Writable {
+			immediate: false
+		});
 
-			const opts = {
+		this[$connection] = new Promise((resolve, reject) => {
+			const connOpts = {
 				host: this.host,
 				port: this.port
 			};
 
-			let connection;
-			let resolved = false;
-
-			if (this.secure) {
-				connection = tls.connect(opts, () => {
-					if (!connection.authorized) {
-						this[$failedAttempts]++;
-						const errMsg = connection.authorizationError;
-						this.emit(new LogentriesError(text.authError(errMsg)));
-					} else if (tls && tls.CleartextStream &&
-						connection instanceof tls.CleartextStream)
-					// Your node is ancient!
-						this.emit('connect');
+			this[$reconnection].on('connect', function (con) {
+				DEBUG.log("Connected");
+				this.emit('connected');
+				con.on('timeout', () => {
+					//we listen to timeout event and close connection if it happens. see below.
+					this.emit("connection timeout");
 				});
-			} else
-				connection = net.createConnection(opts);
-
-			connection.setTimeout(defaults.timeouts.connection.idle);
-
-			// Expose the connection’s drain event
-			connection.on('drain', () =>
-				setImmediate(() => this.emit('connection drain'))
-			);
-
-			// This is an inactivity timeout, not a connection timeout
-			connection.on('timeout', () => {
-				connection.end();
-				delete this[$connection];
+				resolve(con);
 			});
 
-			// Error rejects the promise
-			connection.on('error', () => {
-				connection.destroy();
-				delete this[$connection];
-
-				// If this error occurred before a connection was established, try again.
-				if (!resolved) {
-					this[$getConnection](resolve, reject);
-				}
+			this[$reconnection].on('reconnect', function (n, delay) {
+				if (n > 0)
+					DEBUG.log("Trying to reconnect. Times: " + n + " , previous delay: " + delay);
 			});
 
-			// Close will emit whether it was intended or error-caused:
-			connection.on('close', () => {
-				delete this[$connection];
-
-				this.emit('disconnected');
+			this[$reconnection].on('disconnect', function () {
+				DEBUG.log("Disconnected.");
+				this.emit("disconnected");
 			});
 
-			// This will probably never be triggered since the communication is
-			// one-way, but just in case:
-			connection.on('end', () => {
-				delete this[$connection];
+			this[$reconnection].on('error', function (err) {
+				DEBUG.log("Error occurred during connection: " + err);
 			});
 
-			// Yay
-			connection.on('connect', () => {
-				this[$failedAttempts] = 0;
-				resolved = true;
-				resolve(connection);
+			// now try to connect
+			this[$reconnection].connect(connOpts);
+		});
 
-				this.emit('connected');
-			});
+		this[$reconnection].on('connection timeout', () => {
+			// we owe a lot to inactivity timeout handling with regards to clearing unwanted opened connections hanging around.
+			DEBUG.log("Socket was inactive for a period of time. Destroying.");
+			this.closeConnection();
+			delete this[$connection];
+			this.emit("timed out");
```

Comment:
Not in le_node, but this is for exposing timeout event to the application that uses le_node.